comicapi/genericmetadata.py

```diff
@@ -202,24 +212,122 @@ def get_clean_metadata(self, *attributes: str) -> GenericMetadata:
+        if isinstance(new, list):
+            for a in new:
+                for i, b in enumerate(getattr(self, cur)):
+                    if a.casefold() == b.casefold():
+                        getattr(self, cur).pop(i)
+                        getattr(self, cur).insert(i, a)
+                        break
+        if isinstance(new, set):
+            remove_set = set()
+            for a in new:
+                for b in getattr(self, cur):
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            getattr(self, cur).difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_dedupe_cur(self, new: list[Any] | set[Any], cur: str) -> None:
+        if isinstance(new, list):
+            remove_ids = set()
+            for a in getattr(self, cur):
+                for i, b in enumerate(new):
+                    if a.casefold() == b.casefold():
+                        remove_ids.add(i)
+                        break
+            for r_id in remove_ids:
+                new.pop(r_id)
+            getattr(self, cur).extend(new)
+
+        if isinstance(new, set):
+            remove_set = set()
+            for a in getattr(self, cur):
+                for b in new:
+                    if a.casefold() == b.casefold():
+                        remove_set.add(b)
+                        break
+            new.difference_update(remove_set)
+            getattr(self, cur).update(new)
+
+    def assign_original(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, str) and len(new) == 0:
+                if isinstance(getattr(self, cur), (list, set)):
+                    getattr(self, cur).clear()
+                else:
+                    setattr(self, cur, None)
+            elif isinstance(new, (list, set)) and len(new) == 0:
+                pass
+            else:
+                setattr(self, cur, new)
+
+    def assign_overwrite(self, cur: str, new: Any) -> None:
+        if isinstance(new, (str, list, set)) and len(new) == 0:
+            if isinstance(getattr(self, cur), (list, set)):
+                getattr(self, cur).clear()
+            else:
+                setattr(self, cur, None)
+        else:
+            setattr(self, cur, new)
+
+    def assign_overwrite_no_clear(self, cur: str, new: Any) -> None:
+        if new is not None:
+            if isinstance(new, (str, list, set)) and len(new) > 0:
+                setattr(self, cur, new)
+            elif isinstance(new, (bool, int, float)):
+                setattr(self, cur, new)
+
+    def assign_add_missing(self, cur: str, new: Any) -> None:
+        if new is not None and not getattr(self, cur):
+            setattr(self, cur, new)
+
+    def assign_combine_new(self, cur: str, new: Any) -> None:
+        # Combine lists and sets and prefer 'new' for strings, floats and ints
```

Comment:
So overlay + merge list/set?

Reply:
Yes (mostly?). It will merge lists/sets and replace all other values with `new` but won't clear a list/set on an empty string. Is "merge" the preferred nomenclature?

Reply:
Just wanted to make sure I new what it was trying to do